lib/matplotlib/tight_layout.py

```diff
@@ -327,7 +327,9 @@ def get_tight_layout_figure(fig, axes_list, subplotspec_list, renderer,
         div_row, mod_row = divmod(max_nrows, rows)
         div_col, mod_col = divmod(max_ncols, cols)
         if (mod_row != 0) or (mod_col != 0):
-            raise RuntimeError("")
+            warnings.warn('Incompatible rows or columns in tight_layout; '
```

Comment:
For a better error message I would split the two checks:
~~~
if mod_row != 0:
    warnings.warn('Incompatible rows in tight_layout; tight_layout not applied.') 
if mod_col != 0:
    warnings.warn('Incompatible cols in tight_layout; tight_layout not applied.') 
~~~
If possible even refine the description what is wrong.